### PR TITLE
Add a method to compute the vector tensor product

### PR DESCRIPTION
**Pull request by saddamhijazi**
Add a method to compute the vector tensor product.

### CODE REVIEW

File: src/ITHACA_CORE/EigenFunctions/EigenFunctions.C [hunk 1]
```diff
@@ -64,3 +64,20 @@ void EigenFunctions::sortEigenvalues(Eigen::VectorXd& eigenvalues,
 
     eigenvectors = eigenvectors2;
 }
+
+Eigen::MatrixXd vectorTensorProduct(Eigen::VectorXd g,
```
Comment (giovastabile) [inline src/ITHACA_CORE/EigenFunctions/EigenFunctions.C]: you should pass everything as const reference
for example for g: const Eigen::VectorXd& g

Comment (saddamhijazi) [reply src/ITHACA_CORE/EigenFunctions/EigenFunctions.C]: The method has been modified.

File: src/ITHACA_CORE/EigenFunctions/EigenFunctions.H [hunk 2]
```diff
@@ -221,6 +221,18 @@ class EigenFunctions
...
+        ///
+        /// @brief      A function that computes the product of  g.T c a, where c is a third dim tensor
+        ///
+        /// @param[in]  g     The first vector
+        /// @param[in]  c     The three dim tensor
+        /// @param[in]  a     The second vector
+        ///
+        /// @return     The vector which corresponds to the product g.T c a
+        ///
+        Eigen::MatrixXd vectorTensorProduct(const Eigen::VectorXd& g,
```
Comment (giovastabile) [inline src/ITHACA_CORE/EigenFunctions/EigenFunctions.H:233]: Just one last thing, I think we should do it as template:

template<typename T>
Matrix<T, Eigen::Dynamic, Eigen::Dynamic> vectorTensorProduct(const Matrix<T, Eigen::Dynamic, 1>& g, const Eigen::Tensor<T, 3 >& c, Matrix<T, Eigen::Dynamic, 1>& a);

File: src/ITHACA_CORE/EigenFunctions/EigenFunctions.C [hunk 3]
```diff
@@ -64,3 +64,20 @@ void EigenFunctions::sortEigenvalues(Eigen::VectorXd& eigenvalues,
 
     eigenvectors = eigenvectors2;
 }
+
```
Comment (giovastabile) [inline src/ITHACA_CORE/EigenFunctions/EigenFunctions.C:67]: this one should also be template

File: src/ITHACA_CORE/EigenFunctions/EigenFunctions.H (hunk 2 above)
Comment (saddamhijazi) [reply src/ITHACA_CORE/EigenFunctions/EigenFunctions.H:233]: Do you mean a template with the type being int, double and float ?

Comment (giovastabile) [reply src/ITHACA_CORE/EigenFunctions/EigenFunctions.H:233]: yes, then in the implementation you can leave the general version and add the instantiation of double int etc...

**Comment by giovastabile** [comment]
I have added the function myself and merged in #361. Please @saddamhijazi check that the method is working as expected.

**Pull request closed without merging**